src/Language/Typeside.hs

```diff
@@ -67,12 +67,15 @@ instance (Eq var, Eq ty, Eq sym) => Eq (Typeside var ty sym) where
     = (tys' == tys'') && (syms' == syms'') && (eqs' == eqs'')
 
 instance (Show var, Show ty, Show sym) => Show (Typeside var ty sym) where
-  show (Typeside tys' syms' eqs' _) = "typeside {\n" ++
-    "types\n\t"    ++ intercalate "\n\t" (Prelude.map show $ Set.toList tys') ++
-    "\nfunctions\n\t" ++ intercalate "\n\t" syms'' ++
-    "\nequations\n\t"  ++ intercalate "\n\t" eqs'' ++ " }"
-   where syms'' = Prelude.map (\(k,(s,t)) -> show k ++ " : " ++ show s ++ " -> " ++ show t) $ Map.toList syms'
-         eqs''  = Prelude.map (\(k,s) -> "forall " ++ showCtx k ++ " . " ++ show s) $ Set.toList eqs'
+  show (Typeside tys' syms' eqs' _) =
+    section "typeside" $ unlines
+      [ section "types"     $ unlines . fmap show $ Set.toList tys'
```

Comment:
what do you say of `unlines $ show <$> Set.toList tys'`?

Reply:
I do do that in most places, but this is sort of intentional, as a preparation for a next refactoring pass where I'm going to scoop up the `unlines` and `fmap`, and I wanted to emphasise the presence of the latter a bit.

Reply:
ok, good!